### ISSUE TITLE
RocksDB hits open file descriptor limit

### ISSUE THREAD

**Issue by eyusufatik** [chainwayxyz/citrea#731]
## Summary

On OSX default file descriptor limit is very low, so it's easy to hit that limit.

## Steps to reproduce

1. Sync your node on MacOS, around 500kth block you'll hit the limit (I guess so. this only happened on my machine so far)

## Actual Result
```sh
2024-06-13T07:22:48.093113Z  INFO sov_modules_stf_blueprint: tx #0 hash: 0xa5d2bd846f27861845b0609ce2d8d1cc776b07b807f83747b1ab83b9c7a793c9 result Successful
2024-06-13T07:22:48.103357Z  INFO save_change_set_l2{l2_block_height=474637}: sov_prover_storage_manager: return=()
2024-06-13T07:22:48.103915Z  INFO sov_stf_runner::runner: New State Root after soft confirmation #474637 is: RootHash("6a5f56b8adbfd101fbc8cafee2a88fdd2f98937777b84a73bd9caec50243e368")
2024-06-13T07:22:48.104032Z ERROR finalize_l2{l2_block_height=474637}:finalize_by_l2_height{l2_block_height=474637}: sov_prover_storage_manager: error=IO error: While open a file for appending: full-node-db/state/001016.sst: Too many open files
2024-06-13T07:22:48.104060Z ERROR finalize_l2{l2_block_height=474637}: sov_prover_storage_manager: error=IO error: While open a file for appending: full-node-db/state/001016.sst: Too many open files
2024-06-13T07:22:48.104087Z ERROR sov_stf_runner::runner: error=IO error: While open a file for appending: full-node-db/state/001016.sst: Too many open files
2024-06-13T07:22:48.105350Z ERROR sov_modules_rollup_blueprint: error=IO error: While open a file for appending: full-node-db/state/001016.sst: Too many open files
2024-06-13T07:22:48.105369Z ERROR citrea: Error: IO error: While open a file for appending: full-node-db/state/001016.sst: Too many open files

```
## Additional Information
We might want to look into if we can configure rocks db to use N file descriptors.

**Comment by nkysggsy** [comment]
We can do this. Actually we should use ulimit to set open files to support rocksdb can open enough files. We can use fdlimit::raise_fd_limit. In reth we can see this code.

https://github.com/paradigmxyz/reth/blob/7bfc604a10379b484f636e6041cdf488cc94e7da/crates/node/builder/src/launch/common.rs#L158-L164

```rust
    /// Configure global settings this includes:
    ///
    /// - Raising the file descriptor limit
    /// - Configuring the global rayon thread pool
    pub fn configure_globals(&self) {
        self.inner.configure_globals();
    }
```

https://github.com/paradigmxyz/reth/blob/7bfc604a10379b484f636e6041cdf488cc94e7da/crates/node/builder/src/launch/common.rs#L121-L124
```rust
    pub fn configure_globals(&self) {
        // Raise the fd limit of the process.
        // Does not do anything on windows.
        match fdlimit::raise_fd_limit() {
            Ok(fdlimit::Outcome::LimitRaised { from, to }) => {
                debug!(from, to, "Raised file descriptor limit");
            }
            Ok(fdlimit::Outcome::Unsupported) => {}
            
```
In geth, It have same logic
https://github.com/ethereum/go-ethereum/blob/7cf6a63687da2d9300fec495ff4bc978c4b70b65/cmd/utils/flags.go#L1270-L1293
```go
// MakeDatabaseHandles raises out the number of allowed file handles per process
// for Geth and returns half of the allowance to assign to the database.
func MakeDatabaseHandles(max int) int {
	limit, err := fdlimit.Maximum()
	if err != nil {
		Fatalf("Failed to retrieve file descriptor allowance: %v", err)
	}
	switch {
	case max == 0:
		// User didn't specify a meaningful value, use system limits
	case max < 128:
		// User specified something unhealthy, just use system defaults
		log.Error("File descriptor limit invalid (<128)", "had", max, "updated", limit)
	case max > limit:
		// User requested more than the OS allows, notify that we can't allocate it
		log.Warn("Requested file descriptors denied by OS", "req", max, "limit", limit)
	default:
		// User limit is meaningful and within allowed range, use that
		limit = max
	}
	raised, err := fdlimit.Raise(uint64(limit))
	if err != nil {
		Fatalf("Failed to raise file descriptor allowance: %v", err)
	}
	return int(raised / 2) // Leave half for networking and other stuff
}
```

**Comment by eyusufatik** [comment]
closed by #811
